android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginActivity.kt

```diff
@@ -54,6 +66,16 @@ class LoginActivity :
       configureViews(configurationRegistry.retrieveConfiguration(AppConfigClassification.LOGIN))
     }
 
+    // Check if Pin enabled and stored then move to Pin login
+    val isPinEnabled = loginViewModel.loginViewConfiguration.value?.enablePin ?: false
+    val stayUserNamePasswordLogin =
+      loginViewModel.sharedPreferences.read(FORCE_LOGIN_VIA_USERNAME, false)
+    val lastPinExist = !loginViewModel.sharedPreferences.read(PIN_KEY, EMPTY_STRING).isNullOrEmpty()
```

Comment:
we may want to read/write this preference into secureSharedPref via accountAuthenticator.

Reply:
good catch, will update

Reply:
updated and pushed